raft-tech/TANF-app
# As a developer, I want to improve our backend logging formatter

## Summary of Changes
These changes update our logging characteristics to actually display the logs we've put throughout our application. The root logger was being invoked for the majority of our statements `logger.debug(msg)` and was set to the default level `WARNING` so only warnings/errors would get displayed. These changes create a new logger for our application `tdpservice` and we have set the value such that all children loggers (e.g., `tdpservice.users.api.login`) will inherit all logging properties.

Pull request closes #1707 

## How to Test

1. Choose a dev environment without these changes, and run a tail on the logs:
```
cf login --sso
cf logs tdp-backend-<env name>
```
2. Log into this environment and then log back out and note the logs from our app (or lack thereof). These should mostly consist of the big request messages for user-agent and the very basic "GET" or "POST" logs from django.server.
3. Either switches logging to a new environment with these changes or use a label to deploy this branch to the Cloud.gov environment.
1. Log into and then log back out of this environment and note the differentiation in logging output. Please see screenshots below.


> *Demo GIF(s) and screenshots for testing procedure*
Basics in red before:
![image](https://user-images.githubusercontent.com/84722778/166702318-ded3b065-e5cf-474e-8ecd-36b9130ab803.png)
>
> New additions in both red/white after:
![image](https://user-images.githubusercontent.com/84722778/166703864-2fdf5d5b-a55c-488d-bbbe-642ae588f127.png)

## Tabletop


## Deliverables
_More details on how deliverables herein are assessed included [here](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
- [ ] Logs have code-location information.
- [ ] Logs are visible in cloud.gov's Logstream or accessible in another way.
- [ ] Log files are generated in addition to standard output/error.

### [Deliverable 2: Tested Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
+ Are code coverage minimums met?

### [Deliverable 3: Properly Styled Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)


### [Deliverable 4: Accessible](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)



### [Deliverable 5: Deployed](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)


### [Deliverable 6: Documented](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)


### [Deliverable 7: Secure](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)


### [Deliverable 8: User Research](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
